ros/src/tl_detector/tl_detector.py

```diff
@@ -137,12 +136,20 @@ def find_traffic_lights(self):
             self.state_count = 0
             self.state = state
         elif self._pass_threshold():
-            light_wp = light_wp if state == TrafficLight.RED else -1
             self.last_wp = light_wp
-            self.upcoming_red_light_pub.publish(Int32(light_wp))
+            msg = self._prepare_result_msg(state, light_wp)
+            self.upcoming_red_light_pub.publish(msg)
         else:
-            self.upcoming_red_light_pub.publish(Int32(self.last_wp))
-            self.state_count += 1
+            msg = self._prepare_result_msg(self.state, self.last_wp)
+            self.upcoming_red_light_pub.publish(msg)
+        self.state_count += 1
```

Comment:
Ideally, this line must be under `else` branch unless it causes some issue on your side